consensus/istanbul/core/core.go

```diff
@@ -345,10 +354,41 @@ func (c *core) waitForDesiredRound(r *big.Int) {
 
 func (c *core) updateRoundState(view *istanbul.View, validatorSet istanbul.ValidatorSet, roundChange bool) {
 	// TODO(Joshua): Include desired round here.
-	if roundChange && c.current != nil {
-		c.current = newRoundState(view, validatorSet, nil, c.current.pendingRequest, c.current.preparedCertificate, c.backend.HasBadProposal)
+	if roundChange { // Can a roundChange occur with c.current being nil?
+		var pendingRequest *istanbul.Request
```

Comment:
If we update `newRoundState()` to take `parentSeals` as an argument, could we just do:
```
if roundChange && c.current != nil {
		c.current = newRoundState(view, validatorSet, nil, c.current.pendingRequest, c.current.preparedCertificate, c.current.ParentSeals, c.backend.HasBadProposal)
```
?